apps/core/src/hooks/useElementHeight.ts

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { RefObject, useEffect, useState } from 'react';
+
+export function useElementHeight(elementRef: RefObject<HTMLElement>, defaultHeight: number) {
+	const [height, setHeight] = useState(defaultHeight);
+
+	useEffect(() => {
+		const resizeObserver = new ResizeObserver((entries) => {
```

Comment:
I imagine this is technically slightly wrong because the default height might be wrong without the resize observer ticking